### DISCUSSION TITLE
When (NOT) To use getServerSideProps?

### DISCUSSION THREAD

**Discussion by ripvannwinkler** [vercel/next.js#24581] [Help]
All the documentation says that build time props are better because of static optimization. But assuming I want a "build once / run anywhere" scenario where, let's say for example, I can toggle an analytics component on or off by passing environment variables from the host, it seems to me that **every page** must rely on getServerSideProps to enable accessing the runtime version of the environment variable. Is this correct? If a page doesn't expose a getServerSideProps function, it will instead get whatever the value of that environment variable was at build time, with no way to toggle it by changing the environment variable on the host. Yes?

If my assumptions above are correct, is there a preferred way (that I'm missing) to enable a component to be toggled via host environment variable without exposing a getServerSideProps on every single page?

**Comment by akellbl4** [comment]
You are right if you want to switch some part of the interface immediately. But sometimes a couple or even 5 minutes for an update is okay. It means you can use `getStaticProps` plus `revalidate` time you okay with.

**Reply by ripvannwinkler** [reply]
OMG how did I miss this! The incremental static generation seems kinda buried down the page, but it does exactly what I wanted. Thank you for your input.
